test/gtest/ucp/test_ucp_tag_fallback.cc

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ */
+#include "ucp_test.h"
+#include <ucs/sys/sys.h>
+
+#include <sys/mman.h>
+
+#define MSG_SIZE  (4 * 1024 * ucs_get_page_size())
+
+class test_ucp_tag_fallback : public ucp_test {
+public:
+    void init() {
+        /* forbid zcopy access because it will always fail due to read-only
+         * memory pages (will fail to register memory) */
+        modify_config("ZCOPY_THRESH", "inf");
+        ucp_test::init();
+        sender().connect(&receiver(), get_ep_params());
+        receiver().connect(&sender(), get_ep_params());
+    }
+
+    static ucp_params_t get_ctx_params() {
+        ucp_params_t params = ucp_test::get_ctx_params();
+        params.field_mask  |= UCP_PARAM_FIELD_FEATURES;
+        params.features     = UCP_FEATURE_TAG;
+        return params;
+    }
+};
+
+UCS_TEST_P(test_ucp_tag_fallback, fallback)
+{
+    void *send_buffer;
+    void *recv_buffer;
+    ucs_status_ptr_t send_req;
+    ucs_status_ptr_t recv_req;
+    ucp_request_param_t param = {0};
+
+    /* allocate read-only pages - it force to fail ibv_reg_mr calls */
+    send_buffer = mmap(0, MSG_SIZE, PROT_READ, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+    ASSERT_NE(send_buffer, MAP_FAILED);
+
+    recv_buffer = malloc(MSG_SIZE);
+    ASSERT_NE(recv_buffer, (void*)NULL);
+
+    recv_req = ucp_tag_recv_nbx(receiver().worker(), recv_buffer, MSG_SIZE,
+                                0, 0, &param);
+    ASSERT_UCS_PTR_OK(recv_req);
+
+    send_req = ucp_tag_send_nbx(sender().ep(), send_buffer, MSG_SIZE, 0, &param);
+    ASSERT_UCS_PTR_OK(send_req);
+
+    wait(send_req);
+    wait(recv_req);
+
+    free(recv_buffer);
+    munmap(send_buffer, MSG_SIZE);
+}
+
+#define UCP_INSTANTIATE_TEST_CASE_RNDV_ZCOPY(_test_case) \
+    UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, dcudx, "dc_x") \
+    UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, rc,    "rc_v") \
+    UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, rcx,   "rc_x")
```

Comment:
why not all TLs?

Reply:
test is actual for cases where registration is failed - dc/rc* are good candidates

BTW, knem transport is failed to register on R/O pages